config/routes.rb

```diff
@@ -1,4 +1,27 @@
 Rails.application.routes.draw do
+  root 'questions#index'
+  resources :users
+  resources :questions do
+    resources :answers, only: [:create, :destroy, :edit, :update] do
+      resources :comments, only: [:create, :destroy] do
+        resources :votes, only: [:create,:update]
```

Comment:
As a rule of thumb, don't nest > 2 levels. Also, think about if you need nesting at all. Sometimes hidden params do the job better.